packages/core/src/command/consolidated/consolidated-filter-shared.ts

```diff
@@ -0,0 +1,132 @@
+import { Period, Range } from "@medplum/fhirtypes";
+import { filterTruthy } from "@metriport/shared/common/filter-map";
+import dayjs from "dayjs";
+import customParseFormat from "dayjs/plugin/customParseFormat";
+import isSameOrAfter from "dayjs/plugin/isSameOrAfter";
+import isSameOrBefore from "dayjs/plugin/isSameOrBefore";
+
+dayjs.extend(isSameOrAfter);
+dayjs.extend(isSameOrBefore);
+dayjs.extend(customParseFormat);
+
+export type DateRange = {
+  dateFrom?: string | undefined;
+  dateTo?: string | undefined;
+};
+
+export function arePeriodsWithinRange(
+  periods: (Period | undefined)[] | undefined,
+  range: DateRange
+): boolean | undefined {
+  // we need to check if the periods are valid before we call `.some`, b/c that will always return
+  // boolean and we need to return undefined if there are no valid periods to check - so upstream
+  // functions can check other fields
+  const periodsWithSomeDate = periods?.flatMap(p => (p?.start || p?.end ? p : [])) ?? [];
+  if (periodsWithSomeDate.length < 1) return undefined;
+  return periodsWithSomeDate.some(period => isPeriodWithinRange(period, range));
+}
+export function isPeriodWithinRange(
+  period: Period | undefined,
+  range: DateRange
+): boolean | undefined {
+  const dates = getDatesFromPeriod(period);
+  return areDatesWithinRange(dates, range);
+}
+
+export function areRangesWithinRange(
+  ranges: (Range | undefined)[] | undefined,
+  range: DateRange
+): boolean | undefined {
+  const periodsWithSomeDate = ranges?.flatMap(p => (p?.low || p?.high ? p : [])) ?? [];
```

Comment:
confusing variable naming. dealing with ranges so this should be 
rangesWithSomeDate

Reply:
good catch, copy-pasta